src/views/Project/Details/Users/SearchList/index.js

```diff
@@ -0,0 +1,194 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import SearchInput from '#rsci/SearchInput';
+import ListView from '#rscv/List/ListView';
+import LoadingAnimation from '#rscv/LoadingAnimation';
+
+import {
+    RequestClient,
+    requestMethods,
+} from '#request';
+import { iconNames } from '#constants';
+import _ts from '#ts';
+
+import SearchListItem from './SearchListItem';
+import styles from './styles.scss';
+
+const RequestPropType = PropTypes.shape({
+    pending: PropTypes.bool.isRequired,
+});
+
+const propTypes = {
+    className: PropTypes.string,
+    projectId: PropTypes.number.isRequired,
+    searchInputValue: PropTypes.string.isRequired,
+    onSearchInputChange: PropTypes.func.isRequired,
+    userSearchRequest: RequestPropType.isRequired,
+};
+
+const defaultProps = {
+    className: '',
+};
+
+const emptyList = [];
+const emptyObject = {};
+const MIN_SEARCH_TEXT_CHARACTERS = 2;
+
+const EmptySearch = () => {
+    const emptyText = _ts('project.users', 'searchEmptyText');
+
+    return (
+        <div className={styles.emptyText}>
+            { emptyText }
+        </div>
+    );
+};
+
+const SearchTip = () => {
+    const tipText = _ts(
+        'project.users',
+        'searchTipText',
+        { numberOfCharacters: MIN_SEARCH_TEXT_CHARACTERS },
+    );
+
+    const iconClassName = `
+        ${iconNames.info}
+        ${styles.icon}
+    `;
+    return (
+        <div className={styles.searchTip}>
+            <div className={iconClassName} />
+            <div className={styles.text}>
+                { tipText }
+            </div>
+        </div>
+    );
+};
+
+const requests = {
+    userSearchRequest: {
+        url: '/users-user-groups/',
+        onMount: ({ props: { searchInputValue } }) => {
+            const searchText = searchInputValue.trim();
+            if (searchText.length < MIN_SEARCH_TEXT_CHARACTERS) {
+                return false;
+            }
+
+            return true;
+        },
+        onPropsChanged: {
```

Comment:
I do not understand onPropsChanged attr, so will review this later

Reply:
Can be list of props that will restart the request whenever changed. Or, can be map of functions with prop as key, which will be called when the prop changes. (The function will determine if the request should be restarted, i.e. when the function returns true)